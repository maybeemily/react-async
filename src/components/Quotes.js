import React from 'react';
import PropTypes from 'prop-types';
import Quote from './Quote';

function Quotes({ quoteArray }) {
  const quotes = quoteArray.map((quote, i) => (
    <li key={i}>
      <Quote quote={quote}/>
    </li>
  ));

  return (
    <ul>
      {quotes}
    </ul>
  );
}

Quotes.propTypes = {
  quoteArray: PropTypes.array.isRequired
};

export default Quotes;
