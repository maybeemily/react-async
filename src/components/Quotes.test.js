import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('Quotes component', () => {
  it('renders a list of quotes', () => {
    const testQuotes = [
      {
        character: 'Leela',
        quote: 'Awww',
        image: '/path'
      },
      {
        character: 'Fry',
        quote: 'I.C Weiner??',
        image:'/path'
      }
    ];

    const wrapper = shallow(<Quotes quoteArray={testQuotes}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
