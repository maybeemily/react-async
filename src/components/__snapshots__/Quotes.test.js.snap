// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component renders a list of quotes 1`] = `
<ul>
  <li
    key="0"
  >
    <Quote
      quote={
        Object {
          "character": "Leela",
          "image": "/path",
          "quote": "Awww",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Quote
      quote={
        Object {
          "character": "Fry",
          "image": "/path",
          "quote": "I.C Weiner??",
        }
      }
    />
  </li>
</ul>
`;
